[bitnami/superset] Add chart (#26713)
bitnami/charts: bitnami/superset/README.md

If additional init containers are needed in the same pod, they can be defined using the `initContainers` parameter. Here is an example:

```yaml
initContainers:
  - name: your-image-name
    image: your-image
    imagePullPolicy: Always
    ports:
      - name: portname
        containerPort: 1234
```

Learn more about [sidecar containers](https://kubernetes.io/docs/concepts/workloads/pods/) and [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).

### Pod affinity

This chart allows you to set your custom affinity using the `affinity` parameter. Find more information about Pod affinity in the [kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).

As an alternative, use one of the preset configurations for pod affinity, pod anti-affinity, and node affinity available at the [bitnami/common](https://github.com/bitnami/charts/tree/main/bitnami/common#affinities) chart. To do so, set the `podAffinityPreset`, `podAntiAffinityPreset`, or `nodeAffinityPreset` parameters.

## Persistence

> The [Bitnami Superset](https://github.com/bitnami/containers/tree/main/bitnami/superset) image stores the Superset data and configurations at the `/bitnami` path of the container. Persistent Volume Claims are used to keep the data across deployments.

Author: Actually, Superset does not have persistence, settings are stored in Redis/Postgresql.

[Overview of Apache Superset](https://superset.apache.org/)

Trademarks: This software listing is packaged by Bitnami. The respective trademarks mentioned in the offering are owned by the respective companies, and use of them does not imply any affiliation or endorsement.

## TL;DR

```console
helm install my-release oci://registry-1.docker.io/bitnamicharts/superset
```

Looking to use Superset in production? Try [VMware Tanzu Application Catalog](https://bitnami.com/enterprise), the enterprise edition of Bitnami Application Catalog.

## Introduction

Bitnami charts for Helm are carefully engineered, actively maintained and are the quickest and easiest way to deploy containers on a Kubernetes cluster that are ready to handle production workloads.

This chart bootstraps a [Superset](https://superset.apache.org/) Deployment in a [Kubernetes](https://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.

Bitnami charts can be used with [Kubeapps](https://kubeapps.dev/) for deployment and management of Helm Charts in clusters.

## Prerequisites

- Kubernetes 1.23+
- Helm 3.8.0+
- PV provisioner support in the underlying infrastructure

> - ReadWriteMany volumes for deployment scaling

Reviewer: This chart doesn't have a "persistence" section and the sub-charts (Redis and PostgreSQL) use StatefulSets with PVC Claim Templates, hence I don't think RWX is a requirement:
```suggestion
```

| `externalRedis.username`                     | Redis&reg; username                                                                                                                                                                                                        | `default`          |
| `externalRedis.password`                     | Redis&reg; password                                                                                                                                                                                                        | `""`               |
| `externalRedis.existingSecret`               | Name of an existing secret resource containing the Redis&trade credentials                                                                                                                                                 | `""`               |
| `externalRedis.existingSecretPasswordKey`    | Name of an existing secret key containing the Redis&trade credentials                                                                                                                                                      | `redis-password`   |

> See <https://github.com/bitnami/readme-generator-for-helm> to create the table

Reviewer: Leftover:
```suggestion
```

The above parameters map to the env variables defined in [bitnami/superset](https://github.com/bitnami/containers/tree/main/bitnami/superset). For more information please refer to the [bitnami/superset](https://github.com/bitnami/containers/tree/main/bitnami/superset) image documentation.

Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,

```console
helm install my-release \
  --set auth.username=admin \
  --set auth.password=password \
  --set postgresql.auth.password=secretpassword \
    oci://REGISTRY_NAME/REPOSITORY_NAME/superset
```

> Note: You need to substitute the placeholders `REGISTRY_NAME` and `REPOSITORY_NAME` with a reference to your Helm chart registry and repository. For example, in the case of Bitnami, you need to use `REGISTRY_NAME=registry-1.docker.io` and `REPOSITORY_NAME=bitnamicharts`.

> The above command sets the Superset administrator account username and password to `admin` and `password` respectively. Additionally, it sets the MariaDB `root` user password to `secretpassword`.

Reviewer: Typo:
```suggestion
The above command sets the Superset administrator account username and password to `admin` and `password` respectively. Additionally, it sets the PostgreSQL user password to `secretpassword`.
```
